### PR TITLE
Add a dialog to edit reminders.

### PR DESCRIPTION
**Pull request by gmarty**

### CODE REVIEW

File: app/js/lib/server/reminders.js [hunk 1]
```diff
@@ -41,6 +41,17 @@ export default class Reminders {
   }
 
   /**
+   * Create a new reminder.
+   *
+   * @param {Object} body
+   * @return {Promise}
+   */
+  update(body) {
+    const id = body.id;
```
Comment (arcturus) [inline app/js/lib/server/reminders.js:50]: A couple of comments here, should we check for the existence of `body.id` here and reject.

Also, this method is quite similar to the `set(body)` one, as we send the whole body, should be the api work in that way?

File: app/js/views/reminders/edit-dialog.jsx [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+    this.dueTimeInput = null;
+
+    this.onChangeAction = this.onChangeAction.bind(this);
+    this.onChangeDue = this.onChangeDue.bind(this);
+    this.onSave = this.onSave.bind(this);
+    this.onClose = this.onClose.bind(this);
+  }
+
+  onChangeAction(evt) {
+    const action = evt.target.value;
```
Comment (arcturus) [inline app/js/views/reminders/edit-dialog.jsx]: Will we need validation here?

File: app/js/views/reminders/edit-dialog.jsx [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+
+    const dueMoment = moment()
+      .year(date[0]).month(date[1] - 1).date(date[2])
+      .hour(time[0]).minute(time[1]).second(0);
+    const due = Number(dueMoment.toDate());
+    this.setState({ due });
+  }
+
+  onSave() {
+    const reminder = {
```
Comment (arcturus) [inline app/js/views/reminders/edit-dialog.jsx:48]: could you clone this state?
`const reminger = Object.assign({}, this.state);`

File: app/js/views/reminders/edit-dialog.jsx [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+      action: this.state.action,
+      due: this.state.due,
+    };
+    this.server.reminders.update(reminder)
+      .then(() => {
+        this.refreshReminders();
+        this.hide();
+      })
+      .catch((err) => {
+        console.error(err);
```
Comment (arcturus) [inline app/js/views/reminders/edit-dialog.jsx:60]: Should we refresh the reminders to show the updated info?

File: app/js/views/reminders/reminders-list.jsx [hunk 5]
```diff
@@ -13,12 +14,21 @@ export default class RemindersList extends React.Component {
     };
 
     this.server = props.server;
+    this.refreshReminders = props.refreshReminders;
+
+    this.dialog = null;
```
Comment (arcturus) [inline app/js/views/reminders/reminders-list.jsx]: Should we rename to `editDialog` just to avoid clash with any future extra dialogs in the list?

**Comment by arcturus** [comment]
LGTM r+, just left some comments about naming.

File: app/js/lib/server/reminders.js (hunk 1 above)
Comment (gmarty) [inline app/js/lib/server/reminders.js:50]: I'll check for the existence of id.

The other difference with `set` is the HTTP verb (POST vs. PUT).

File: app/js/views/reminders/edit-dialog.jsx (hunk 2 above)
Comment (gmarty) [inline app/js/views/reminders/edit-dialog.jsx]: Good idea!

File: app/js/views/reminders/edit-dialog.jsx (hunk 3 above)
Comment (gmarty) [inline app/js/views/reminders/edit-dialog.jsx:48]: I'm not sure I understand. What should I do with the cloned state?

File: app/js/views/reminders/edit-dialog.jsx (hunk 4 above)
Comment (gmarty) [inline app/js/views/reminders/edit-dialog.jsx:60]: In this case, the update failed so the original reminder is left unchanged.

File: app/js/views/reminders/reminders-list.jsx (hunk 5 above)
Comment (gmarty) [inline app/js/views/reminders/reminders-list.jsx]: Good point. Will do!

File: app/js/views/reminders/edit-dialog.jsx (hunk 3 above)
Comment (arcturus) [inline app/js/views/reminders/edit-dialog.jsx:48]: Just to build the reminder object in 1 line, but just realise that you have more attributes in a state object.

File: app/js/views/reminders/edit-dialog.jsx (hunk 4 above)
Comment (arcturus) [inline app/js/views/reminders/edit-dialog.jsx:60]: Ok :)

**Pull request merged**